### DISCUSSION TITLE
PKCS12 file samples

### DISCUSSION THREAD

**Discussion by timlegge** [openssl/openssl#24764] [Q&A]
Hi

I have been updating perl's Crypt::OpenSSL::PKCS12 to support the -info of 'openssl pkcs12' command.  I have replicated the output and I have also pulled the output into a perl hash.  Everything seems to work ok and I have test samples of files that include:
 1. NID_pkcs8ShroudedKeyBag
 2. NID_safeContentsBag
 3. NID_certBag

From my google searches it appears that the following are not widely used.
 1. NID_keyBag
 2. NID_secretBag
    
Is there a way to generate files for these or do you you know of any samples so I can test the perl module's output.

Tim

**Comment by mattcaswell** [comment]
> NID_keyBag

Calling the pkcs12 app with "-keypbe NONE" should do this I think

> NID_secretBag

See:

https://github.com/openssl/openssl/blob/7afa7731e924d5ac10fc992d8cd777f407d33af9/test/pkcs12_format_test.c#L595-L623

**Reply by timlegge** [reply]
Thanks @mattcaswell NID_keyBag works now and I will look at the other
